### PR TITLE
chore: reorder and use wallet name from lud16

### LINKED ISSUE #126: Change to not allow creating wallet without a valid connection string
Currently the app architecture creates a wallet entry and then it is updated once you scan/paste a NWC URL. This doesn't really make sense because it can create new wallets that do not have a connection string and just need to be deleted again.

See https://github.com/getAlby/go/issues/92

### LINKED ISSUE #127: Default to use wallet name from LUD16
See https://github.com/getAlby/go/issues/92

Rather than having to enter a wallet name before creating the wallet, it should be done afterwards and could be pre-filled with the lightning address.

### LINKED ISSUE #164: [Bug] Wallet Profile Cannot Be Deleted if Not Connected to NWC
**Description:**  
A wallet profile becomes inaccessible if no NWC QR code is scanned during setup, preventing users from deleting it.

**Steps to Reproduce:**  
1. Create a new wallet profile.
2. Do not scan an NWC QR code when prompted.
3. Try to access the wallet profile settings to delete it.

**Expected Behavior:**  
Users should be able to access the wallet settings and delete the profile, even if no NWC QR code was scanned.

**Actual Behavior:**  
The wallet profile remains inaccessible, blocking deletion.

**Suggested Solution:**  
Consider allowing users to bypass the QR code prompt and access the settings directly or implement a fallback option for deletion.

**Impact:**  
Users unable to manage or remove unused wallet profiles, leading to frustration.

**Additional Context:**  
Reported by user feedback on Alby Go. A temporary workaround involves assigning a dummy QR code to access the settings and delete the wallet.  

We have a video from a user showing this, it is on our slack, on the Alby Go channel.

### PR DESCRIPTION
**Pull request by im-adithya**
Fixes #126
Fixes #127
Fixes #164

## Delete wallets with faulty/no NWC url (#164)
<table>
<tr>
<td>
<h3>
We can now select empty wallets without being redirected to setup
</h3>
</td>

<td>
<h3>
That makes wallet setting page accessible
</h3>
</td>

<td>
<h3>
Wallet can be deleted from there
</h3>
</td>
</tr>
<tr>
<td>

</td>

<td>

</td>

<td>

</td>
</tr>
</table>

## Don't create wallet without valid connection string (#126)
<table>
<tr>
<td>
<h3>
Connect a new wallet (no. of wallets: 1)
</h3>
</td>

<td>
<h3>
If nwc url is invalid it doesn't create a wallet</h3>
</td>

<td>
<h3>
Go back and check (no. of wallets: Still 1)
</h3>
</td>
</tr>
<tr>
<td>

</td>

<td>

</td>

<td>

</td>
</tr>
</table>

## Reordering and using LUD16 (#127)
<table>
<tr>
<td>
<h3>
Go to setup and scan/paste nwc string
</h3>
</td>

<td>
<h3>
LUD 16 is used as name from string
</h3>
</td>

<td>
<h3>
If LUD 16 is not available
</h3>
</td>

<td>
<h3>
After finishing setup
</h3>
</td>
</tr>

<tr>
<td>

</td>

<td>

</td>

<td>

</td>

<td>

</td>
</tr>
</table>

### CODE REVIEW

**Comment by rolznz** [comment]
I believe this does not truly fix the root issue described in https://github.com/getAlby/go/issues/126 which is that we shouldn't even save a new wallet until it's correctly configured. Would it be good to implement this too?

File: lib/state/appStore.ts [hunk 1]
```diff
@@ -171,18 +169,6 @@ export const useAppStore = create<AppState>()((set, get) => {
...
-        nostrWalletConnectUrl: undefined,
-      });
-
-      set({ nwcClient: undefined });
-    },
-    setNostrWalletConnectUrl: (nostrWalletConnectUrl) => {
-      updateCurrentWallet({
-        nostrWalletConnectUrl,
-      });
-    },
```
Comment (im-adithya) [inline lib/state/appStore.ts:185]: These aren't beign used anywhere

**Comment by reneaaron** [comment]
Just tested and it's working out great :muscle: 

The only thing that imo should be changed is the order of steps:

You start on the Connect Wallet screen where you scan a QR code. If the Connection Secret contains a LUD16 identifier that one should be used as the wallet's name, otherwise you can enter one. 

If you don't enter a name (or close the app) the connection should not be saved yet.

**Review by reneaaron — APPROVED** [review]
tACK

**Pull request merged**